segmentation/utils/spatial_analysis.py

```diff
@@ -0,0 +1,272 @@
+import pandas as pd
+import xarray as xr
+import numpy as np
+import scipy
+import statsmodels
+from statsmodels.stats.multitest import multipletests
+
+# Erin's Data Inputs
+
+# cell_array = pd.read_csv("/Users/jaiveersingh/Downloads/SpatialEn"
+#                          "richment/granA_cellpheno_CS-asinh-norm_revised.csv")
+# marker_thresholds = pd.read_csv("/Users/jaiveersingh/Downloads/Sp"
+#                                 "atialEnrichment/markerThresholds.csv")
+# dist_matrix = np.asarray(pd.read_csv("/Users/jaiveersingh/Documen"
+#                                      "ts/MATLAB/distancesMat5.csv",
+#                                      header=None))
+
+
+def compute_close_cell_num(patient_data, patient_data_markers, thresh_vec,
+                           dist_mat, marker_num, dist_lim, cell_label_idx):
+    """Finds positive cell labels and creates matrix with counts for cells positive for corresponding markers.
+
+    This function loops through all the included markers in the patient data and identifies cell labels positive for
+    corresponding markers. It then subsets the distance matrix to only include these positive cells and records
+    interactions based on whether cells are close to each other (within the dist_lim). It then stores the number of
+    interactions in the index of close_num corresponding to both markers (for instance markers 1 and 2 would be in
+    index [0, 1]).
+
+    Args:
+        patient_data: cell expression data for the specific point
+        patient_data_markers: cell expression data of markers
+            for the specific point
+        thresh_vec: matrix of thresholds column for markers
+        dist_mat: cells x cells matrix with the euclidian
+            distance between centers of corresponding cells
+        marker_num: number of markers in expresion data
+        dist_lim: threshold for spatial enrichment distance proximity
+        cell_label_idx: column with cell labels
+
+    Returns:
+        close_num: marker x marker matrix with counts for cells
+            positive for corresponding markers
+        marker1_num: list of number of cell labels for marker 1
+        marker2_num: list of number of cell labels for marker 2"""
+
+    # Create close_num, marker1_num, and marker2_num
+    close_num = np.zeros((marker_num, marker_num), dtype='int')
+    marker1_num = []
+    marker2_num = []
+
+    for j in range(0, marker_num):
+        # Identify cell labels that are positive for respective markers
+        marker1_thresh = thresh_vec.iloc[j]
+        marker1posinds = patient_data_markers[patient_data_markers.columns[j]] > marker1_thresh
+        marker1poslabels = patient_data.loc[marker1posinds, patient_data.columns[cell_label_idx]]
+        marker1_num.append(len(marker1poslabels))
```

Comment:
marker1posinds is a boolean array, so its length would still be 80 for example, whereas the length of marker1poslabels is only the labels with TRUE, so it would only be 12. Should I instead replace it with like marker1_num.append(sum(marker1posinds))